test/model/src/driver/debug_log_parser.cpp

```diff
@@ -309,6 +398,41 @@ debug_log_parser::apply(kv_transaction_ptr txn, const txn_timestamp &op)
     txn->set_commit_timestamp(op.commit_ts);
 }
 
+/*
+ * debug_log_parser::begin_transaction --
+ *     Begin a transaction.
+ */
+kv_transaction_ptr
+debug_log_parser::begin_transaction(const debug_log_parser::commit_header &op)
+{
+    if (_base_write_gen == k_write_gen_none)
+        throw model_exception("The base write generation is not set");
+
+    kv_transaction_ptr txn = _database.begin_transaction();
+    txn->set_wt_metadata(op.txnid, _base_write_gen);
+    return txn;
+}
+
+/*
+ * debug_log_parser::commit_transaction --
+ *     Commit/finalize a transaction.
+ */
+void
+debug_log_parser::commit_transaction(kv_transaction_ptr txn)
```

Comment:
The presence of this made me realise there's something I don't understand about the way this model hangs together - how do rollbacks/aborts work? Does there need any code to handle them?

Reply:
The model itself handles rollbacks (it's all in `kv_transaction`), but in the case of feeding the model from the debug log, we don't have to concern ourselves with them, because WiredTiger logs only committed transactions.

Rollbacks will become more important when we get to the next phase of the project, in which we will generate workload ourselves.

Reply:
Got it - thanks for explaining that.